util/econftool.c

```diff
@@ -279,15 +286,29 @@ int main (int argc, char *argv[])
         char path_tmpfile_edit[PATH_MAX];
         char path_tmpfile_orig[PATH_MAX];
         char *TMPPATH = getenv("TMPDIR");
+        char *tmpName = NULL;
         if (TMPPATH == NULL)
             TMPPATH = "/tmp";
 
-        snprintf(path_tmpfile_edit, sizeof(path_tmpfile_edit), "%s", tmpnam(NULL));
-        snprintf(tmpfile_edit, sizeof(tmpfile_edit), "%s",
-                strrchr(path_tmpfile_edit, '/') + 1);
-        snprintf(path_tmpfile_orig, sizeof(path_tmpfile_orig), "%s", tmpnam(NULL));
-        snprintf(tmpfile_orig, sizeof(tmpfile_orig), "%s",
-                strrchr(path_tmpfile_orig, '/') + 1);
+        tmpName = tmpnam(NULL);
+        if (tmpName != NULL) {
+            snprintf(path_tmpfile_edit, sizeof(path_tmpfile_edit), "%s", tmpName);
+            snprintf(tmpfile_edit, sizeof(tmpfile_edit), "%s",
+                    strrchr(path_tmpfile_edit, '/') + 1);
+            tmpName = NULL;
+        } else {
+            fprintf(stderr, "Couldn't generate tmpfile name\n");
+            exit(EXIT_FAILURE);
+        }
+        tmpName = tmpnam(NULL);
+        if (tmpName != NULL) {
+            snprintf(path_tmpfile_orig, sizeof(path_tmpfile_orig), "%s", tmpName);
+            snprintf(tmpfile_orig, sizeof(tmpfile_orig), "%s",
+                    strrchr(path_tmpfile_orig, '/') + 1);
+        } else {
+            fprintf(stderr, "Couldn't generate tmpfile name\n");
+            exit(EXIT_FAILURE);
+        }
```

Comment:
Sorry, I just noticed that it may be even better to use `tempnam` (instead of `tmpnam`) here because it is possible to define a prefix (e.g. `econftool-`).
This will reduce potential file conflicts and allow easier attribution for the user when looking through the files in /tmp - for example in case the application was interrupted and wasn't able to clean up its files. Additionally the `TMPPATH` will be determined by the function (but it has to be split later for the _libeconf_ call).

The error handling may be adjusted to print the `errno` then. (I admit only one errno is possible in this case, butit's still good practice...)